### ISSUE TITLE
rank_zero_only removal violate backward compatibillity

### ISSUE THREAD

**Issue by moghadas76** [Lightning-AI/lightning#15509]
cc @borda @justusschock

**Comment by carmocca** [comment]
Can you provide more details of what broke exactly and in which version? We might have made a mistake. In that case we can add it back in a bug-fix release.

**Comment by visoutre** [comment]
hi @carmocca this issue seems to be related to several issues with scripts hosted by CompVis which import rank_zeo_only, so I'll add some info rather than open a new issue. 

After the last update (sometime around Nov 1st), importing rank_zero_only from 'pytorch_lightning.utilities.distributed' was causing an ImportError. from pytortch_lightning/utilities/distributed.py

A temporary solution is importing using this line instead:
`from pytorch_lightning.utilities.rank_zero import rank_zero_only`

I'm not sure what the permanent solution is, this Github stuff is new to me. It would be nice to get a suggestion from you guys or to bug fix in the next update. [More details on this problem are documented here](https://github.com/AUTOMATIC1111/stable-diffusion-webui/discussions/4111#discussioncomment-4075026)

**Comment by carmocca** [comment]
Hi @visoutre. Thanks for the description.

I've opened a PR making the modification to the CompVis repository: https://github.com/CompVis/taming-transformers/pull/190. You can also make this edit in your local installation to unblock you using 1.8.0 in your work.

These functions had been deprecated since version 1.6.0, to be removed in 1.8.0. It's unfortunate that this had not been updated before. To alleviate the confusion, I have opened a PR here to provide the user with a better message on what to do if this happens: #15569

**Comment by visoutre** [comment]
@carmocca Sounds good, thank you!

Is a pull request something I can do as well? [CompVis has 1 more script using the old import](https://github.com/CompVis/stable-diffusion/blob/main/ldm/models/diffusion/ddpm.py)

**Comment by carmocca** [comment]
I left comments in an exisitng PR for the stable diffusion repo: https://github.com/CompVis/stable-diffusion/pull/451

**Comment by moghadas76** [comment]
Thanks you guys
